Content.Goobstation.Shared/Weapons/Multishot/SharedMultishotSystem.cs

```diff
@@ -32,104 +51,161 @@ public override void Initialize()
         SubscribeLocalEvent<MultishotComponent, GotEquippedHandEvent>(OnEquipWeapon);
         SubscribeLocalEvent<MultishotComponent, GotUnequippedHandEvent>(OnUnequipWeapon);
         SubscribeLocalEvent<MultishotComponent, GunRefreshModifiersEvent>(OnRefreshModifiers);
-        SubscribeLocalEvent<MultishotComponent, GunShotEvent>(OnShotAttempt);
+        SubscribeLocalEvent<MultishotComponent, GunShotEvent>(OnGunShot);
+        SubscribeLocalEvent<MultishotComponent, AmmoShotEvent>(OnAmmoShot);
+        SubscribeLocalEvent<MultishotComponent, ExaminedEvent>(OnExamined);
+        SubscribeAllEvent<RequestShootEvent>(OnRequestShoot);
     }
 
-    private void OnShotAttempt(Entity<MultishotComponent> multishotWeapon, ref GunShotEvent args)
+    private void OnRequestShoot(RequestShootEvent msg, EntitySessionEventArgs args)
     {
-        var comp = multishotWeapon.Comp;
-
-        if (comp.RelatedWeapon == null)
-            return;
+        var user = args.SenderSession.AttachedEntity;
 
-        // Need to prevent recursive shoot attempting
-        if (_handsSystem.GetActiveItem(args.User) != multishotWeapon)
+        if (user == null ||
+            !_combatSystem.IsInCombatMode(user))
             return;
 
-        if (!TryComp<GunComponent>(comp.RelatedWeapon, out var relatedGun) || !TryComp<GunComponent>(multishotWeapon, out var gun))
-            return;
+        var gunsEnumerator = GetMultishotGuns(user.Value);
+        var shootCoords = GetCoordinates(msg.Coordinates);
+        var target = GetEntity(msg.Target);
 
-        if (gun.ShootCoordinates == null)
-            return;
+        foreach(var gun in gunsEnumerator)
+        {
+            var (gunEnt, gunComp, _) = gun;
 
-        if (TryComp(comp.RelatedWeapon.Value, out GunComponent? otherGun))
-            otherGun.Target = gun.Target;
+            if (!HasComp<MultishotComponent>(GetEntity(msg.Gun)) && gunEnt != GetEntity(msg.Gun))
+                continue;
 
-        _gunSystem.AttemptShoot(args.User, comp.RelatedWeapon.Value, relatedGun, gun.ShootCoordinates.Value);
+            if (gunComp.Target == null || !gunComp.BurstActivated || !gunComp.LockOnTargetBurst)
+                gunComp.Target = target;
 
-        // Synchronizing reload timer
-        var reloadDelta = relatedGun.LastFire - gun.LastFire;
-        relatedGun.NextFire -= reloadDelta.Duration();
+            _gunSystem.AttemptShoot(user.Value, gunEnt, gunComp, shootCoords);
+        }
     }
 
-    private void OnRefreshModifiers(Entity<MultishotComponent> multishotWeapon, ref GunRefreshModifiersEvent args)
+    private void OnGunShot(Entity<MultishotComponent> multishotWeapon, ref GunShotEvent args)
```

Comment:
```suggestion
    private void OnGunShot(EntityUid uid, MultishotComponent comp, ref GunShotEvent args)
```
again if were not gonna use the Entity<T> pattern for its use then lets not use it at all